src/linear_algebra/basic.lean

```diff
@@ -1641,6 +1641,19 @@ rfl
 
 end prod
 
+section curry
+
+variables (V V₂ R)
+
+/-- Linear equivalence between an uncurried and curried function. -/
```

Comment:
Please add a reference to [tensor_product.curry](https://leanprover-community.github.io/mathlib_docs/linear_algebra/tensor_product.html#tensor_product.curry) in the docstring. This is a different operation but a person looking for it may find your definition first.

Usually we also add lemmas `coe_curry` and `coe_curry_symm`.